src/client/hooks/useMyPeer.tsx

```diff
@@ -1,64 +1,229 @@
 import { useCallback, useEffect, useState } from "react";
-import Peer from "peerjs";
-import { cleanup } from "@testing-library/react";
+import Peer, { MediaConnection } from "peerjs";
+import { Map } from "immutable";
+import { useMediaStream } from "./useMediaStream";
+import { PeerData } from "../peer";
+import { VideoEvent } from "../../events";
+import { UserPeer, VideoPeersResponseType } from "../../types";
+import { useFetch } from "./useFetch";
+import { requestIsLoaded } from "../utils";
 
 declare var process: any;
 
 export type PeerId = string;
 export type Peers = Map<PeerId, Peer.MediaConnection>;
+const options: Peer.PeerJSOption = {
+    host: "/",
+    path: "/peerServer",
+    secure: process.env.NODE_ENV === "production",
+    port: process.env.NODE_ENV === "production" ? 443 : 5000,
+};
 
-export const useMyPeer: () => [Peer | undefined, PeerId, () => void] = () => {
-    const options: Peer.PeerJSOption = {
-        host: "/",
-        path: "/peerServer",
-        secure: process.env.NODE_ENV === "production",
-        port: process.env.NODE_ENV === "production" ? 443 : 5000,
-    };
-
-    const [myPeer, setMyPeer] = useState<Peer | undefined>(
-        () => new Peer(options)
-    );
+export const useMyPeer = (
+    socket: SocketIOClient.Socket,
+    userId: string,
+    sessionId: string
+): PeerData => {
+    const [myPeer] = useState<Peer>(() => new Peer(options));
     const [myPeerId, setMyPeerId] = useState<PeerId>("");
+    const [peerCalls, setPeerCalls] = useState<Map<string, MediaConnection>>(
+        Map()
+    );
+    const [peerStreams, setPeerStreams] = useState<Map<string, MediaStream>>(
+        Map()
+    );
+    const [response] = useFetch<
+        VideoPeersResponseType,
+        { sessionId: string; userId: string }
+    >("/videos/peers", { sessionId, userId });
+    const [myStream, enableStream, disableStream] = useMediaStream();
     const cleanUp = useCallback(() => {
         if (myPeer) {
             myPeer.disconnect();
+            setPeerStreams(peerStreams.clear());
+            peerCalls.forEach((call) => {
+                call.close();
+            });
+            setPeerCalls(peerCalls.clear());
             myPeer.destroy();
-            setMyPeer(undefined);
+            disableStream();
             console.log("Cleaning up");
         }
-    }, [myPeer]);
+    }, [disableStream, myPeer, peerCalls, peerStreams]);
 
-    const setupPeer = useCallback(() => {
-        console.log("Setup peer");
-        if (!myPeer) {
-            setMyPeer(new Peer(options));
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    // const setupPeer = useCallback(() => {
```

Comment:
Consider removing the comments if they are no longer used.

Reply:
Most of the comments in your reviews are "backup code" that might be useful for future use. They are basically alternative ways of doing things in case something goes bad. I'd refrain from removing those, and I'll try to remove anything I can

Reply:
Sounds good, just a reminder 👍